Update protobuf dependency

TalweSingh opened the pull request.

TalweSingh committed:
Update protobuf-related dependencies

Signed-off-by: Talwinder Singh <talwinder.singh@frequenz.com>

TalweSingh committed:
Update `RELEASE_NOTES.md`

Signed-off-by: Talwinder Singh <talwinder.singh@frequenz.com>

shsms commented on pyproject.toml:
```diff
@@ -37,11 +37,11 @@ dependencies = [
   # We can't widen beyond 6 because of protobuf cross-version runtime guarantees
   # https://protobuf.dev/support/cross-version-runtime-guarantee/#major
-  "protobuf >= 4.25.3, < 6", # Do not widen beyond 6!
+  "protobuf >= 6.31.1, < 8", # Do not widen beyond 6!
```
```suggestion
  "protobuf >= 6.31.1, < 8", # Do not widen beyond 8!

shsms reviewed (commented):
Just one comment

TalweSingh committed:
Add pytest filterwarnings to ignore protobuf version warnings

This change is also proposed in frequenz-repo-config but has not yet been released.

Signed-off-by: Talwinder Singh <talwinder.singh@frequenz.com>

TalweSingh committed:
Update `frequenz-api-common` dependency

Signed-off-by: Talwinder Singh <talwinder.singh@frequenz.com>

shsms commented on pyproject.toml:
```diff
@@ -33,15 +33,15 @@ classifiers = [
-  "frequenz-api-common >= 0.6.0, < 0.7.0",
+  "frequenz-api-common >= 0.8.0, < 0.9.0",
   "googleapis-common-protos >= 1.56.4, < 2",
   # We can't widen beyond 6 because of protobuf cross-version runtime guarantees
```
Damn, too many mentions of version numbers.


```suggestion
  # We can't widen beyond 8 because of protobuf cross-version runtime guarantees
```

shsms commented on pyproject.toml:
```diff
@@ -33,15 +33,15 @@ classifiers = [
 requires-python = ">= 3.11, < 4"
 dependencies = [
-  "frequenz-api-common >= 0.6.0, < 0.7.0",
+  "frequenz-api-common >= 0.8.0, < 0.9.0",
```
Best to mention this in the release notes as well, under Upgrading

shsms reviewed (commented):
2 more \o/

TalweSingh committed:
Update `googleapis-common-protos`

Signed-off-by: Talwinder Singh <talwinder.singh@frequenz.com>
